src/execution_plan/ops/shared/create_functions.c

```diff
@@ -52,6 +52,7 @@ static void _CommitNodes(PendingCreations *pending) {
 		if(n->label != NULL) {
 			s = GraphContext_GetSchema(gc, n->label, SCHEMA_NODE);
 			assert(s);
+			n->labelID = s->id; // Update the label ID within the node.
```

Comment:
we set labelID but we do not set label, how come?

Reply:
If the query introduced the label, as in:
`CREATE (a:Person {name: 'a'}) RETURN a`
The QGNode label string will be `Person`, and the label ID will be `GRAPH_UNKNOWN_LABEL` (-2).

The new label will be added at https://github.com/RedisGraph/RedisGraph/blob/master/src/execution_plan/ops/shared/create_functions.c#L36-L37
We perform the ID update here because if we performed it at the time of schema creation, it would only update the first QGNode with that label in the query.

Reply:
Right, but for every node created don't we need to set `n->label` in addition to `n->labelID` ?

Reply:
No, `n->label` is populated for every node from the AST string. If it weren't populated, we wouldn't be here, as the condition on 52 is `if(n->label != NULL)`.